mysite/ct/tests.py

```diff
@@ -691,3 +691,22 @@ def test_continue_partial(self, url_name, url_post_data):
         self.assertFalse(PartialHashTable.objects.filter(token=token).exists())
         result = self.client.get(continue_url)
         self.assertEqual(result.status_code, 404)
+
+    def test_get_on_anonymous_user(self):
+        """
+        Test for PartialAction for anonymous user.
+        """
+        response = self.client.post(
+            reverse('ct:partial_pause'),
+        )
+        continue_url = json.loads(response.content).get('partial_url')
+        self.client.logout()
+        result = self.client.get(continue_url)
+        self.assertContains(result, 'User is not authenticated', status_code=401)
+
+
```

Comment:
We need a test where there is no record DB (wrong token) as well as with existent token.
Something like your previous test:

    def test_get_on_anonymous_user_wrong_token(self):
        """
        Test that checking for Anonymous is started before query to DB.
        """
        result = self.client.get(reverse('ct:partial_continue', kwargs={'token': 'wrong_token'}))
        self.assertContains(result, 'User is not authenticated', status_code=401)

Even more - we can check DB query using mock:

    @patch('ct.views.get_object_or_404')
    def test_get_on_anonymous_user_wrong_token(self, get_object_or_404):
        """
        Test that checking for Anonymous is started before query to DB.
        """
        result = self.client.get(reverse('ct:partial_continue', kwargs={'token': 'wrong_token'}))
        self.assertContains(result, 'User is not authenticated', status_code=401)
        self.assertEqual(get_object_or_404.call_count, 0)  # checking that there are no queries to DB using get_object_or_404